psydac/linalg/solvers.py

```diff
@@ -145,14 +147,15 @@ def _check_options(self, **kwargs):
                 raise ValueError(f"Key '{key}' not understood. See self._options for allowed keys.")
 
     def transpose(self):
-        At = self._A.T
+        At = self._A.H
         solver = self._solver
         options = self._options
         return inverse(At, solver, **options)
```

Comment:
We hesitate with @jowezarek, because the classical transposition with complex matrices should be the Hermitian transpose (so the H). Thus we decide that the transpose problem in complex should be with A.H and not A.T.

Reply:
I understand, but the function name `transpose` does not seem right to me!

Reply:
Maybe we can call it `H_transpose` for the hermitian transpose.

[c4acb03](https://github.com/pyccel/psydac/pull/273/commits/c4acb034953c1d44e3fe90606dd98fc1680e0120)

Reply:
In fact, each solver should have a `transpose` method. So, we should have the `transpose` function define as Julian define it before. Thus, we should have a second method `H_transpose`.

But I don't think that the `tranpose` method, defined by Julian, have a meaning with complex value.

Reply:
I agree that we need two separate functions. At a minimum we need the two properties `T` and `H`.

Reply:
In this case, this commit create the function `H` for each solvers and redefine `transpose`. Since T is defined in `LinearOperator`, those 3 functions work.
[8e27ebc](https://github.com/pyccel/psydac/pull/273/commits/8e27ebca33851f265204cab9eee0ba9c76549829)

Also, this commit add some tests for the functions `transpose` and `H` .
[f16e641](https://github.com/pyccel/psydac/pull/273/commits/f16e6416fd162f22d5c675468160039fd13bfc5d)

Reply:
As it appears that we do not need `LinearOperator` to have the `conjugate` method, I think the cleanest solution is to have only one method called `transpose` which accepts the `conjugate` argument:

```python
@abstractmethod
def transpose(self, *, conjugate=False, out=None):
    pass

@property
def T(self):
    return self.transpose()

@property
def H(self):
    return self.transpose(conjugate=True)
```
@tomcaruso @jowezarek: What do you think?

Reply:
I agree. It would be cleaner for the `LinearOperator` subclasses contained in the folder `linalg`.